### ISSUE TITLE
Filters do not work with dependencies tool on Windows

### ISSUE THREAD

**Issue by HertzDevil** [crystal-lang/crystal#14923]
The `--include` and `--exclude` options for `crystal tool dependencies` have no effect on Windows. For example, running `bin\crystal tool dependencies src\prelude.cr -isrc` at the repository root outputs nothing, whereas running `crystal tool dependencies src\shards.cr -esrc` at the Shards root doesn't suppress any output.

This is most likely because the include and exclude patterns are expanded into native paths:

https://github.com/crystal-lang/crystal/blob/1baf3a726f76adda630d5ee4f384dd00e319a2de/src/compiler/crystal/tools/dependencies.cr#L11-L12

They are then passed to `File.match?`:

https://github.com/crystal-lang/crystal/blob/1baf3a726f76adda630d5ee4f384dd00e319a2de/src/compiler/crystal/tools/dependencies.cr#L112-L114

which only recognizes the forward slash as the directory separator, not the backslash.

**Comment by straight-shoota** [comment]
Yeah, `crystal tool unreachable` uses the same filtering mechanism and there we're converting the paths to the posix type.
I'm not sure this is 100% correct because there might be some odd edge cases. But it should work just fine for relevant use cases.

I've meant to combine and abstract the filtering logic in a helper library, including proper specs. This should be happening sometime.
